main.py

```diff
@@ -0,0 +1,193 @@
+from fastapi import FastAPI, HTTPException
+from pydantic import BaseModel
+from sqlalchemy import create_engine, Column, Integer, String, ForeignKey
+from sqlalchemy.orm import relationship, sessionmaker
+from sqlalchemy.ext.declarative import declarative_base
+
+# Создаем экземпляр FastAPI
+app = FastAPI()
+
+# Создаем экземпляр базы данных
+engine = create_engine('sqlite:///shop.db', echo=True)
+Base = declarative_base()
+
+# Определяем модели данных
+class User(Base):
+    __tablename__ = 'users'
+    id = Column(Integer, primary_key=True)
+    first_name = Column(String)
+    last_name = Column(String)
+    email = Column(String)
+    password = Column(String)
+
+    orders = relationship('Order', back_populates='user')
+
+class Order(Base):
+    __tablename__ = 'orders'
+    id = Column(Integer, primary_key=True)
+    user_id = Column(Integer, ForeignKey('users.id'))
+    product_id = Column(Integer, ForeignKey('products.id'))
+    order_date = Column(String)
+    status = Column(String)
+
+    user = relationship('User', back_populates='orders')
+    product = relationship('Product', back_populates='orders')
+
+class Product(Base):
+    __tablename__ = 'products'
+    id = Column(Integer, primary_key=True)
+    name = Column(String)
+    description = Column(String)
+    price = Column(Integer)
+
+    orders = relationship('Order', back_populates='product')
+
+# Создаем таблицы в базе данных
+Base.metadata.create_all(engine)
+
+# Создаем сессию для взаимодействия с базой данных
+Session = sessionmaker(bind=engine)
+session = Session()
+
+# Определяем модели данных Pydantic для валидации запросов и ответов
+class UserRequest(BaseModel):
+    first_name: str
+    last_name: str
+    email: str
+    password: str
+
+class UserResponse(BaseModel):
+    id: int
+    first_name: str
+    last_name: str
+    email: str
+
+class OrderRequest(BaseModel):
+    user_id: int
+    product_id: int
+    order_date: str
+    status: str
+
+class OrderResponse(BaseModel):
+    id: int
+    user_id: int
+    product_id: int
+    order_date: str
+    status: str
```

Comment:
Модели практически идентичны, а значит одна может наследоваться от другой.
```suggestion
class OrderResponse(OrderRequest):
    id: int
```